### PR TITLE
Comments

### PR DESCRIPTION
**Pull request by drutyper**
Ive coded in the the render method to show the comments in the post/show

I also changed the comments/_form a bit  so that it can work properly. 

But I am running into an error whenever I click on comments

```NoMethodError in Posts#show
Showing /Users/Dru/code/Bloccit/app/views/comments/_form.html.erb where line #2 raised:

undefined method `topic_post_comments_path' for #<#Class:0x007f8b008cbc60:0x007f8b031afd20>
Extracted source (around line #2):
1
2
3
4
5

  <% if current_user %>
  <%= form_for [topic, post, comment], remote: true do |f| %>
    <% if comment.errors.any? %>
      <div class="alert alert-block">
        <h4>There are <%= pluralize(comment.errors.count, "error") %> errors.</h4>

Trace of template inclusion: app/views/posts/show.html.erb

```
```

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -1,11 +1,13 @@
...
 
+
   devise_for :users
   resources :users, only: [:update]
    
    resources :topics do
-     resources :posts, except: [:index]
-   end
+   resources :posts, except: [:index]
+   resources :comments, only: [:create]
```
Comment (Dagnan) [inline config/routes.rb:9]: comments should be nested under posts, and posts should be nested under topics. This is why you have your error I guess.

File: app/views/comments/_form.html.erb [hunk 2]
```diff
@@ -0,0 +1,36 @@
+<% if current_user %>
+<%= form_for [topic, post, comment], remote: true do |f| %>
```
Comment (Dagnan) [inline app/views/comments/_form.html.erb]: Why are you using `remote: true`?

File: config/routes.rb (hunk 1 above)
Comment (Dagnan) [inline config/routes.rb:9]: See the generated routes with `rake routes` and see if you have something like 'topic_post_comments'.

Comment (drutyper) [inline config/routes.rb:9]: When you say nest under post, do you mean like this: resources :post, :comments, [:index]
and topics would look like this: resources :topics, posts do

Edit: I tried both ways but did not work, so I changed line 7 to: 
resources :topics, :posts, :comments do

File: app/views/comments/_form.html.erb (hunk 2 above)
Comment (drutyper) [inline app/views/comments/_form.html.erb]: I was trying different things so I tried this, but it didn't make a difference. I'm changing it back

File: config/routes.rb [hunk 3]
```diff
@@ -1,11 +1,13 @@
...
+
   devise_for :users
   resources :users, only: [:update]
    
-   resources :topics do
-     resources :posts, except: [:index]
-   end
+   resources :topics, :posts, :comments do
+   resources :posts, except: [:index]
+   resources :comments, only: [:create]
```
Comment (Dagnan) [inline config/routes.rb:9]: Ok so posts is nested under topics, but comments needs to be nested under posts as well.

**Comment by Dagnan** [comment]
I see you merged the pull requests. Is everything working as you want?

File: config/routes.rb (hunk 3 above)
Comment (drutyper) [inline config/routes.rb:9]: I tried to do resources :posts, :comments except: [:index]
But then I run into an error when I run rails s

/Users/Dru/.rvm/gems/ruby-2.1.5/gems/activesupport-4.1.6/lib/active_support/dependencies.rb:241:in `load': /Users/Dru/code/Bloccit/config/routes.rb:8: syntax error, unexpected tIDENTIFIER, expecting keyword_end (SyntaxError)
   resources :posts, :comments except: [:index]

**Comment by drutyper** [comment]
I did that by mistake because I tried to do a pull request again for my the changes I made but I dont think I did it right. Because I dont want to merge to master, just copy over the comments branch with the new changes.

File: config/routes.rb (hunk 3 above)
Comment (Dagnan) [inline config/routes.rb:9]: No. posts is nested under topics, this is what you have:

```
resources :topics do
  resources :posts, except: [:index]
 end
```

Just nest comments under posts:

```
resources :topics do
  resources :posts, except: [:index] do
    resources :comments, only: [:create]
  end
 end
```

Don't forget to run `rake routes` and adjust accordingly.

Comment (Dagnan) [inline config/routes.rb:9]: Please read again http://guides.rubyonrails.org/routing.html. You really must read it.

Comment (Dagnan) [inline config/routes.rb:9]: Also, the nesting part (Nesting Posts) on this checkpoint: https://www.bloc.io/users/andrew-liwanag/checkpoints/311

**Pull request merged**